beacon-chain/rpc/validator/proposer.go

```diff
@@ -293,19 +293,37 @@ func (vs *Server) deposits(ctx context.Context, currentVote *ethpb.Eth1Data) ([]
 		return []*ethpb.Deposit{}, nil
 	}
 
-	upToEth1DataDeposits := vs.DepositFetcher.AllDeposits(ctx, canonicalEth1DataHeight)
-	depositData := [][]byte{}
-	for _, dep := range upToEth1DataDeposits {
-		depHash, err := ssz.HashTreeRoot(dep.Data)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not hash deposit data")
+	var depositTrie *trieutil.SparseMerkleTrie
+
+	finalizedDeposits := vs.DepositFetcher.FinalizedDeposits(ctx)
```

Comment:
This block of logic should be its own helper func for easier testing and scoping

Reply:
+1